Lesson6.ipynb

```diff
@@ -27,7 +34,686 @@
+    "\n",
+    "Each Object has its own attributes, and behavior. Objects are separate from one another. They have their own existence, their own identity that is independent of other objects.\n",
+    "\n",
+    "For example, if you consider the Biggest Loser challenge, each member can be an object with a name, age, gender as its attributes and whether active participant or not can be its behavior.\n",
+    "\n",
+    "### Well, how do we construct this objects in our program? Class!\n",
+    "\n",
+    "### What is Class?\n",
+    "Well, a class is a place where you can identify the behaviors and properties of an Object. So, the properties and behavior of an Object will be defined inside a class.\n",
+    "\n",
+    "The syntax of the class definition:\n",
+    "\n",
+    "```python\n",
+    "\tclass ClassName:\n",
+    "\t\t'Optional class documentation string'\n",
+    "\t\tclass_suite\n",
+    "```\n",
+    "> *The class has a documentation string referred to a class variable ```__doc__```, which can be accessed via ```ClassName.__doc__```*\n",
+    "\n",
+    "> *The class_suite consists of all the component statements defining class members, data attributes, and functions.*\n",
+    "\n",
+    "Let's try it!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Member:\n",
+    "    \"Its a member class\"        #documentation string\n",
+    "    size = 0                    #Class variable\n",
+    "    def __init__(self, name, age, isCaptain, totalStepCount):    #initialize method called when class object is created\n",
+    "        self.name = name\n",
+    "        self.age = age\n",
+    "        self.isCaptain = \"Yes\" if isCaptain else \"No\"       #assign \"Yes\" if the isCaptian value is True else assigns \"No\"\n",
+    "        self.totalStepCount = totalStepCount\n",
+    "        Member.totalMembers += 1                            #totalMembers keeps increases by 1, everytime a new Member object is created"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The variable *`size`* is a class variable whose value is shared among all instances of this class. This can be accessed as *`Member.size`* from inside or outside the class.\n",
+    "\n",
+    "The first method ***`__init__()`*** is a special method, which is called class constructor or initialization method that Python calls when you create a new instance of this class.\n",
+    "\n",
+    "You declare other class methods like normal functions with the exception that the first argument to each method is ***`self`***. Python adds the ***`self`*** argument to the list for you; you do not need to include it when you call the methods.\n",
+    "\n",
```

Comment:
Would be nice to have an example of class method right here. The next example is after a new section.
Maybe just this as example:
```python
class Member:
    'Holds the details of a team member'                    #documentation string
    totalMembers = 0                                        #Class variable

    def __init__(self, name, age, teamName, isCaptain, totalStepCount):    #initialize method called when class object is created
        self.name = name
        self.age = age
        self.teamName = teamName
        self.isCaptain = "Yes" if isCaptain else "No"       # assign "Yes" if the isCaptain value is True else assigns "No"
        self.totalStepCount = totalStepCount
        Member.totalMembers += 1                            # totalMembers is increased by 1 every time a new instance of Member is created

    def printDetails(self):
        print("Member:", self.name, " Team:", self.teamName, " Total Steps:", self.totalStepCount, "Captain:", getattr(self,"isCaptain"), "\n")
                                                            # getattr() method here returns the value of isCaptain. it's an alias for self.isCaptain 
```